Update README docs (#2417)
wasp-lang/wasp: waspc/README.md

is a convenient way of installing and selecting versions of `cabal`, `hls` and
`ghc`.

> :warning: If you are on a Mac, we still recommend you use the official [ghcup](https://www.haskell.org/ghcup/) installer over Homebrew, as the initial setup will be easier, and you won't have to worry about setting the correct environment variables.

Author: I've added this bit, because I tried to install ghcup through Homebrew, and the env vars were all wrong.

> :warning: If you are on a Mac, we still recommend you use the official [ghcup](https://www.haskell.org/ghcup/) installer over Homebrew, as the initial setup will be easier, and you won't have to worry about setting the correct environment variables.

Reviewer: ```suggestion
:warning: On Mac, we recommend using the official [ghcup](https://www.haskell.org/ghcup/) installer over Homebrew, as it works out of the box.
```

Nice! A small suggestion, since the details about env vars is a bit out of blue / potentially overly specific.

In [cabal.project](cabal.project) we're explicitly saying what compiler version
to build Wasp with.

```cabal
with-compiler: ghc-x.y.z
```

> Please ensure that you have this ghc version installed.

Author: Note: I removed the link here, because I found it confusing. I expected that it would take me to a page with the correct version of GHC, but it actually took me to their semantic version policy.

Reviewer: Makes sense, I wonder why we put this link in the first place! Doesn't seem to be very helpful here.
